src/attesters/tdx-ecdsa/collect_endorsements.c

```diff
@@ -0,0 +1,81 @@
+/* Copyright (c) 2021 Intel Corporation
+ * Copyright (c) 2020-2023 Alibaba Cloud
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <string.h>
+#include <unistd.h>
+#include <rats-tls/log.h>
+#include <rats-tls/attester.h>
+#include <stddef.h>
+#include <tdx_attest.h>
+#include <sgx_dcap_quoteverify.h>
+
+enclave_attester_err_t tdx_ecdsa_collect_endorsements(enclave_attester_ctx_t *ctx,
+						      attestation_evidence_t *evidence,
+						      attestation_endorsement_t *endorsements)
+{
+	enclave_attester_err_t ret = ENCLAVE_ATTESTER_ERR_NONE;
+	sgx_ql_qve_collateral_t *collateral = NULL;
+
+	RTLS_DEBUG("ctx %p, evidence %p, endorsements %p\n", ctx, evidence, endorsements);
+
+	uint32_t collateral_size;
+	quote3_error_t qv_ret = tee_qv_get_collateral(evidence->tdx.quote, evidence->tdx.quote_len,
+						      (uint8_t **)&collateral, &collateral_size);
+	if (SGX_QL_SUCCESS != qv_ret) {
+		RTLS_ERR("tee_qv_get_collateral(): 0x%04x\n", qv_ret);
+		ret = SGX_ECDSA_ATTESTER_ERR_CODE((int)qv_ret);
+		goto err;
+	}
+
+	/* Copy fields from collateral to endorsements->ecdsa */
+	sgx_ql_qve_collateral_t *c = collateral;
```

Comment:
Why need to declare `sgx_ql_qve_collateral_t *c`?

Reply:
Same reason as above, this is to make `COPY_ENDORSEMENT_FIELD` look more concise